Add zap to CIPD setup

tpudlik opened the pull request:
This makes the tool available for running the tests in CI.

@andreilitvin

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/project-chip/connectedhomeip?pullRequest=24630) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/project-chip/connectedhomeip?pullRequest=24630) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/project-chip/connectedhomeip?pullRequest=24630) it.</sub>

bzbarsky-apple commented on scripts/zap.json:
```diff
@@ -0,0 +1,9 @@
+        {
+            "path": "fuchsia/third_party/zap/${platform}",
+            "platforms": ["linux-amd64", "linux-arm64", "mac-amd64", "windows-amd64"],
+            "tags": ["version:2@v2023.01.19-nightly.1"]
```
This is going to need to change often.  Please update the zap version update script so it updates this too.

tpudlik replied:
Done! Thanks for pointing this out.

andy31415 reviewed (approved):
Approved: had offline chat about this - CIPD is automated to get the latest release of zap within 4 hours ... this seems to actually speed up our development cycle because we stop requiring a separate docker file.

tpudlik commented:
@andy31415 I updated the zap update script, but will leave any Dockerfile and documentation updates to you.

tpudlik commented:
I believe the only blocker is that I need approval from another reviewer.

@bzbarsky-apple could you approve (assuming I've addressed your concerns!)?
